services/brig/src/Brig/API/Client.hs

```diff
@@ -438,3 +453,41 @@ removeLegalHoldClient uid = do
   -- maybe log if this isn't the case
   forM_ legalHoldClients (execDelete uid Nothing)
   wrapHttpClient $ Intra.onUserEvent uid Nothing (UserLegalHoldDisabled uid)
+
+createAccessToken ::
+  (Member JwtTools r, Member Now r, Member PublicKeyBundle r) =>
+  UserId ->
+  ClientId ->
+  StdMethod ->
+  Link ->
+  Maybe Proof ->
+  ExceptT CertEnrollmentError (AppT r) (DPoPAccessTokenResponse, CacheControl)
+createAccessToken uid cid method link = \case
+  Nothing -> throwE ProofMissing
+  Just proof -> do
+    domain <- Opt.setFederationDomain <$> view settings
+    nonce <- ExceptT $ note NonceNotFound <$> wrapClient (Nonce.lookupAndDeleteNonce uid (cs $ toByteString cid))
+    httpsUrl <- do
+      let urlBs = "https://" <> toByteString' domain <> "/" <> cs (toUrlPiece link)
+      maybe (throwE MisconfiguredRequestUrl) pure $ fromByteString $ urlBs
+    maxSkewSeconds <- Opt.setDpopMaxSkewSecs <$> view settings
+    expiresIn <- Opt.setDpopTokenExpirationTimeSecs <$> view settings
+    now <- fromUTCTime <$> lift (liftSem Now.get)
+    let expiresAt = now & addToEpoch expiresIn
+    pathToKeys <- ExceptT $ note KeyBundleError . Opt.setPublicKeyBundle <$> view settings
+    pubKeyBundle <- ExceptT $ note KeyBundleError <$> liftSem (PublicKeyBundle.get pathToKeys)
```

Comment:
Well, I can't get it to work, probably because of the `Sem` effect from `PublicKeyBundle.get`.

And for the other places, `domain` and `expiresIn` are used again later, so that doesn't work either.